lib/vault/api/auth_token.rb

```diff
@@ -24,7 +24,12 @@ class AuthToken < Request
     #
     # @return [Secret]
     def create(options = {})
-      json = client.post("/v1/auth/token/create", JSON.fast_generate(options))
+      url = if options.include?(:role)
+              "/v1/auth/token/create/" + options.delete(:role)
+            else
+              "/v1/auth/token/create"
+            end
+      json = client.post(url, JSON.fast_generate(options))
```

Comment:
Haha, yep. I realized this in the middle of the night. D'oh.

I also would have expected it to have a `!` at the end as it modifies the calling object. Anyways! Would love to see this merged.